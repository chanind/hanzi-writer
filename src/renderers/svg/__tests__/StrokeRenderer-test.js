const yi = require('hanzi-writer-data/一.json');
const StrokeRenderer = require('../StrokeRenderer');
const RenderTarget = require('../RenderTarget');
const parseCharData = require('../../../parseCharData');


const char = parseCharData('一', yi);

describe('StrokeRenderer', () => {

  let target;

  beforeEach(() => {
    document.body.innerHTML = '<div id="target"></div>';
    target = RenderTarget.init('target');
  });

  it('renders a path and clipPath', () => {
    const props = {
      strokeColor: {r: 12, g: 101, b: 20, a: 0.3},
      radicalColor: null,
      strokeWidth: 2,
      opacity: 0.7,
      displayPortion: 0.4,
    };

    const renderer = new StrokeRenderer(char.strokes[0]);
    renderer.mount(target, props);
    renderer.render(props);

    expect(target.defs.childNodes.length).toBe(1);
    expect(target.defs.childNodes[0].nodeName).toBe('clipPath');
    const maskPath = target.defs.childNodes[0].childNodes[0];
    expect(maskPath.nodeName).toBe('path');
    const maskId = target.defs.childNodes[0].getAttribute('id');
    expect(target.svg.childNodes.length).toBe(2); // defs and path
    expect(target.svg.childNodes[1].nodeName).toBe('path');
    expect(target.svg.childNodes[1].getAttribute('stroke')).toBe('rgba(12,101,20,0.3)');
    expect(target.svg.childNodes[1].getAttribute('clip-path')).toBe(`url(https://test.com/url#${maskId})`);

    expect(maskPath).toMatchSnapshot();
  });

});
