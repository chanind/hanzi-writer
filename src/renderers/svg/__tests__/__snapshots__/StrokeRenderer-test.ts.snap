// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StrokeRenderer renders a path and clipPath 1`] = `
<path
  d="M 518 518 Q 572 515 623 511 Q 758 501 900 517 Q 928 521 935 510 Q 944 495 930 481 Q 896 448 845 425 Q 829 418 798 427 Q 723 440 480 466 Q 180 491 137 492 Q 130 492 124 492 Q 108 492 106 505 Q 105 520 127 537 Q 146 552 183 566 Q 195 570 216 562 Q 232 556 306 546 Q 400 527 518 518 Z"
/>
`;
